addons/l10n_sa/models/account_journal.py

```diff
@@ -0,0 +1,38 @@
+from odoo import api, models, _
+from odoo.exceptions import UserError
+
+
+class AccountJournal(models.Model):
+    _inherit = 'account.journal'
+
+    @api.model
+    def _fill_missing_values(self, vals):
+        super()._fill_missing_values(vals)
+        journal_type = vals.get('type')
+        company = self.env['res.company'].browse(vals['company_id']) if vals.get('company_id') else self.env.company
+        random_account = self.env['account.account'].search([('company_id', '=', company.id)], limit=1)
+
+        digits = len(random_account.code) if random_account else 6
+        if journal_type in ('bank', 'cash'):
+            has_payment_accounts = vals.get('payment_debit_account_id') or vals.get('payment_credit_account_id')
```

Comment:
payment_debit_account_id and payment_credit_account_id don't exist anymore.
Now, there are fields for that on the company and on payment methods. By default, the account selected is the one from the company. The payment method may define one (not mandatory), in order to change that. 

So ... do we still need an override here ?

Reply:
Indeed, it is no longer needed. I removed it